### PR TITLE
proxy requests from 3002 to 3001 so that the UI only talks to itself

### PR DESCRIPTION
**Pull request by attwad**
Yes I lied when I said I was out for today, couldn't let that one go sorry.

This is great, this means all /api calls are now made to the same origin. It will make auth much simpler.

I had to customize the proxy as described in https://create-react-app.dev/docs/proxying-api-requests-in-development/#configuring-the-proxy-manually because one cannot use "localhost" in docker composed networks (https://docs.docker.com/compose/networking/).

### CODE REVIEW

File: verification/curator-service/ui/Dockerfile [hunk 1]
```diff
@@ -20,5 +20,4 @@ EXPOSE 3002
 # Start service, will trigger a build.
 # Environment variables must be passed, notably:
 # - PORT
-# - REACT_APP_API_ENDPOINT
```
Comment (axmb) [inline verification/curator-service/ui/Dockerfile:23]: This is now REACT_APP_PROXY_URL, right?

File: verification/curator-service/ui/package.json [hunk 2]
```diff
@@ -26,7 +27,7 @@
   },
   "scripts": {
     "start-noenv": "react-scripts start",
-    "start": "PORT=3002 REACT_APP_API_ENDPOINT=http://localhost:3001 react-scripts start",
+    "start": "PORT=3002 react-scripts start",
```
Comment (axmb) [inline verification/curator-service/ui/package.json:30]: If you're in the business of `|| <default>` for the API endpoint, do you want to just do the same for the port?

File: verification/curator-service/ui/src/setupProxy.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const { createProxyMiddleware } = require('http-proxy-middleware');
```
Comment (axmb) [inline verification/curator-service/ui/src/setupProxy.js:1]: Should probably `.ts` this, given that it's in src.

File: verification/curator-service/ui/Dockerfile (hunk 1 above)
Comment (attwad) [reply verification/curator-service/ui/Dockerfile:23]: ah yeah sorry I did that before adding a default value that doesn't work with docker, added back.

File: verification/curator-service/ui/src/setupProxy.js (hunk 3 above)
Comment (attwad) [reply verification/curator-service/ui/src/setupProxy.js:1]: Actually no, the docs say that this should be named setupProxy.js.
react automatically calls that if it exists, I am not sure we can make it a typescript thing considering app is being passed as a param and comes from "somewhere" that I couldn't find out.

**Comment by attwad** [comment]
[Set up a login and auth mechanism.](https://trello.com/c/YKmdRRjW/97-set-up-a-login-and-auth-mechanism)

**Pull request merged**